### PR TITLE
violently rearranges some flags and instance vars

### PR DESCRIPTION
**Pull request by Spookerton**
ref commits

notable extra: adds /abstract_type Initialize behavior, allowing types to specify themselves as abstract by self-reference in order to prevent occult instances. Currently lacks implementations; followups use it.

### CODE REVIEW

File: code/game/objects/items/weapons/explosives.dm [hunk 1]
```diff
@@ -126,4 +126,4 @@
 				blast_flash = blast_light + round(new_blast_power * 0.75)
 			else
 				to_chat(user, "<span class='notice'>The [I] is not any better than the component already installed into this charge!</span>")
-	return .
\ No newline at end of file
+	return .
```
Comment (Atermonera) [inline code/game/objects/items/weapons/explosives.dm:129]:
```suggestion
```
This is the curse of me reviewing PRs that touch a hundred somewhat-random files.

File: code/modules/reagents/machinery/dispenser/cartridge.dm [hunk 2]
```diff
@@ -55,10 +55,10 @@
 	..()
 	if (is_open_container())
 		to_chat(usr, "<span class = 'notice'>You put the cap on \the [src].</span>")
-		flags ^= OPENCONTAINER
+		atom_flags ^= ATOM_REAGENTS_IS_OPEN
 	else
 		to_chat(usr, "<span class = 'notice'>You take the cap off \the [src].</span>")
-		flags |= OPENCONTAINER
+		atom_flags |= ATOM_REAGENTS_IS_OPEN
```
Comment (Atermonera) [inline code/modules/reagents/machinery/dispenser/cartridge.dm:61]: This pains me

**Review by Atermonera — COMMENTED** [review]
Just a bunch of minor crap, I'll merge this tomorrow

File: code/game/area/areas.dm [hunk 3]
```diff
@@ -13,12 +13,17 @@
 	plane = PLANE_LIGHTING_ABOVE //In case we color them
 	luminosity = 0
 	mouse_opacity = 0
+
+	var/area_flags = 0
```
[the remark this answers is not in the document]
Comment (Spookerton) [reply code/game/area/areas.dm:17]: I'll cover *rather a few* of these in another pass with less complexity

File: code/_onclick/adjacent.dm [hunk 4]
```diff
@@ -114,14 +114,14 @@ Quick adjacency (to turf):
...
-	This is defined as any dense ON_BORDER object, or any dense object without throwpass.
+	This is defined as any dense ATOM_HAS_TRANSITION_PRIORITY object, or any dense object without throwpass.
 	The border_only flag allows you to not objects (for source and destination squares)
 */
 /turf/proc/ClickCross(var/target_dir, var/border_only, var/target_atom = null)
 	for(var/obj/O in src)
 		if( !O.density || O == target_atom || O.throwpass) continue // throwpass is used for anything you can click through
 
-		if( O.flags&ON_BORDER) // windows have throwpass but are on border, check them first
+		if( O.atom_flags&ATOM_HAS_TRANSITION_PRIORITY) // windows have throwpass but are on border, check them first
```
[the remark this answers is not in the document]
Comment (Spookerton) [reply code/_onclick/adjacent.dm:124]: A curse of demons past

File: code/game/objects/items/weapons/explosives.dm (hunk 1 above)
Comment (Spookerton) [reply code/game/objects/items/weapons/explosives.dm:129]: `return .` is actually the /tg/ favored way of implying returning the default return value (even though return does that anyway). Lummox updated byond to remove an extra instruction just so this pattern wasn't detrimental.

Although I think it's stylistically detrimental. 🥴

File: code/modules/reagents/machinery/dispenser/cartridge.dm (hunk 2 above)
Comment (Spookerton) [reply code/modules/reagents/machinery/dispenser/cartridge.dm:61]: And there's more to come!

Just, easier, since just rename-in-place.

Comment (Atermonera) [reply code/modules/reagents/machinery/dispenser/cartridge.dm:61]: Moreso the `if(condition) print(text) else print(basically the same text)`

**Pull request merged**